WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCardViewModelSlice.kt

```diff
@@ -0,0 +1,54 @@
+package org.wordpress.android.ui.mysite.cards.dashboard.bloganuary
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import org.wordpress.android.ui.bloggingprompts.BloggingPromptsSettingsHelper
+import org.wordpress.android.ui.mysite.MySiteCardAndItemBuilderParams.BloganuaryNudgeCardBuilderParams
+import org.wordpress.android.ui.mysite.SiteNavigationAction
+import org.wordpress.android.util.config.BloganuaryNudgeFeatureConfig
+import org.wordpress.android.viewmodel.Event
+import javax.inject.Inject
+
+class BloganuaryNudgeCardViewModelSlice @Inject constructor(
+    private val bloganuaryNudgeFeatureConfig: BloganuaryNudgeFeatureConfig,
+    private val bloggingPromptsSettingsHelper: BloggingPromptsSettingsHelper,
+) {
+    private val _onNavigation = MutableLiveData<Event<SiteNavigationAction>>()
+    val onNavigation = _onNavigation as LiveData<Event<SiteNavigationAction>>
+
+    private lateinit var scope: CoroutineScope
+
+    fun initialize(scope: CoroutineScope) {
+        this.scope = scope
+    }
+
+    fun getBuilderParams(): BloganuaryNudgeCardBuilderParams {
+        // TODO thomashortadev: check if current device date is in December 2023
```

Comment:
Same question about the TODOs in this class (if they should be removed or if they will be handled by another PR)

Also I wonder if it's possible for us to use the server date instead.

Reply:
The blogging prompt logic to get the current-day prompt uses the local device date, so it makes sense to use the device date here as well as that's when the prompt will change anyway.

The TODOs I left in code were intentional since the feature is in-development and there are a few empty methods.